aiida_lsmo/workchains/isotherm_accurate.py

```diff
@@ -540,8 +541,25 @@ def should_run_another_gcmc_highp(self):
         self.ctx.pressure_old = self.ctx.pressure
         self.ctx.pressure = self.ctx.pressure_old + min(self.ctx.dpmax, dp_computed)
 
-        if self.ctx.pressure > self.ctx.psat:
-            return False  # Converged
+        if self.ctx.gcmc_loading_average > 0.5 * self.ctx.geom['Estimated_saturation_loading']:
+            # Compute slope of straight line
+            preg1 = list(self.ctx.raspa_gcmc[-3].outputs.output_parameters['framework_1']['components'].values())[0]['partial_pressure']/100000
+            preg2 = list(self.ctx.raspa_gcmc[-2].outputs.output_parameters['framework_1']['components'].values())[0]['partial_pressure']/100000
+            preg3 = list(self.ctx.raspa_gcmc[-1].outputs.output_parameters['framework_1']['components'].values())[0]['partial_pressure']/100000
+            conv1 = list(self.ctx.raspa_gcmc[-3].outputs.output_parameters['framework_1']['components'].values())[0]['conversion_factor_molec_uc_to_mol_kg']
+            conv2 = list(self.ctx.raspa_gcmc[-2].outputs.output_parameters['framework_1']['components'].values())[0]['conversion_factor_molec_uc_to_mol_kg']
+            conv3 = list(self.ctx.raspa_gcmc[-1].outputs.output_parameters['framework_1']['components'].values())[0]['conversion_factor_molec_uc_to_mol_kg']
+            load1 = list(self.ctx.raspa_gcmc[-3].outputs.output_parameters['framework_1']['components'].values())[0]['loading_absolute_average']
+            load2 = list(self.ctx.raspa_gcmc[-2].outputs.output_parameters['framework_1']['components'].values())[0]['loading_absolute_average']
+            load3 = list(self.ctx.raspa_gcmc[-1].outputs.output_parameters['framework_1']['components'].values())[0]['loading_absolute_average']
+
+            slope = scipy.stats.linregress([preg1, preg2, preg3], [conv1*load1, conv2*load2, conv3*load3])[0]
+            inter = scipy.stats.linregress([preg1, preg2, preg3], [conv1*load1, conv2*load2, conv3*load3])[1]
+
+            self.ctx.pressure = (self.ctx.geom['Estimated_saturation_loading'] - inter)/slope
+            self.ctx.gcmc_1 += 1
```

Comment:
Should that be
```suggestion
            self.ctx.gcmc_i += 1
```
?

Reply:
I am sorry, it was a typo but it changes nothing !!
I am trying something now and will let you know if it works.
In the outline of the workchain, a new gcmc calculation is called only if the While loop returns True.
The fact that we have return False there then it converges and exits.
I switched it to True and added a dummy condition as False afterwards.

Reply:
still not working :(

Reply:
Is the gcmc at least running now?

Could you commit & push your changes?